lib/url/URLModulesPlugin.js

```diff
@@ -0,0 +1,50 @@
+/*
+	MIT License http://www.opensource.org/licenses/mit-license.php
+	Author Tobias Koppers @sokra
+*/
+
+"use strict";
+
+const URLGenerator = require("./URLGenerator");
+const URLParser = require("./URLParser");
+
+const JavascriptModulesPlugin = require("../JavascriptModulesPlugin");
+
+class URLModulesPlugin {
+	constructor() {
+		this.plugin = {
+			name: "URLModulesPlugin"
+		};
+	}
+
+	apply(compiler) {
+		const { plugin } = this;
+		const { compilation } = compiler.hooks;
+
+		compilation.tap(plugin, (compilation, { normalModuleFactory }) => {
+			const { createParser, createGenerator } = normalModuleFactory.hooks;
+
```

Comment:
Here you need to hook into `renderManifest` and render out the modules as assets, similar to the WebAssemblyModulesPlugin.

Reply:
We need some logic to generate asset name from module. Best if this is configurable and hookable.